docs/features/mcp-enhancements.md

```
@@ -0,0 +1,1569 @@
+    tool-cache,
+    request-batcher,
+    elicitation,
+    server-base,
+    annotations,
+    multi-server,
+    registry,
+  ]
+---
+
+# MCP Enhancements
+
```

markdownlint-cli2 (0.21.0)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)